farming/logics/traits/master_chef/farming.rs

```diff
@@ -524,11 +510,12 @@ pub trait Farming: Storage<Data> + Storage<ownable::Data> + FarmingGetters + Far
                 arsw_reward = arsw_reward
                     .checked_add(
                         casted_mul(
-                            (self._period_max(period)? as u128)
+                            (self
+                                ._period_max(period)?
                                 .checked_sub(last_block.into())
                                 .ok_or(FarmingError::SubUnderflow5)?
-                                .checked_mul(pool_info.alloc_point.into())
-                                .ok_or(FarmingError::MulOverflow6)?,
+                                * pool_info.alloc_point)
+                                .into(),
```

Comment:
We should cast multiplicands to u128, as suggested at https://github.com/AstarNetwork/wasm-showcase-dapps/pull/16#discussion_r1009141189, to prevent unnecessary overflow.